pkg/kubelet/kubelet.go

```diff
@@ -428,7 +428,7 @@ func NewMainKubelet(
 			imageBackOff,
 			serializeImagePulls,
 			enableCustomMetrics,
-			klet.hairpinMode == componentconfig.HairpinVeth,
+			klet.hairpinMode == componentconfig.HairpinVeth && networkPluginName != "kubenet",
```

Comment:
sure but right now i'll break if we abruptly do that for plugins that don't support the flag, because the docker runtime is handling it for them